diesel/src/result.rs

```diff
@@ -279,43 +279,43 @@ impl StdError for Error {
     }
 
     fn cause(&self) -> Option<&StdError> {
-        match *self {
-            Error::InvalidCString(ref e) => Some(e),
-            Error::QueryBuilderError(ref e) => Some(&**e),
-            Error::DeserializationError(ref e) => Some(&**e),
-            Error::SerializationError(ref e) => Some(&**e),
+        match self {
+            Error::InvalidCString(e) => Some(e),
+            Error::QueryBuilderError(e) => Some(&**e),
```

Comment:
I was surprised that I had to keep `&**` here. I would have expected to be able to remove at least one `*`.